lib/modules/manager/gradle/parser.ts

```diff
@@ -1,1207 +1,907 @@
-        }
-        dep.managerData = { fileReplacePosition, packageFile };
-      }
-      return { deps: [dep] };
-    }
-  }
-  return null;
-}
-
-function processPlugin({
-  tokenMap,
-  packageFile,
-  variables,
-}: SyntaxHandlerInput): SyntaxHandlerOutput {
-  const { pluginName, pluginVersion, methodName } = tokenMap;
-  const plugin = pluginName.value;
-  const depName =
-    methodName.value === 'kotlin' ? `org.jetbrains.kotlin.${plugin}` : plugin;
-  const packageName =
-    methodName.value === 'kotlin'
-      ? `org.jetbrains.kotlin.${plugin}:org.jetbrains.kotlin.${plugin}.gradle.plugin`
-      : `${plugin}:${plugin}.gradle.plugin`;
-
-  const dep: PackageDependency<GradleManagerData> = {
-    depType: 'plugin',
-    depName,
-    packageName,
-    registryUrls: ['https://plugins.gradle.org/m2/'],
-    commitMessageTopic: `plugin ${depName}`,
-  };
-
-  if (pluginVersion.type === TokenType.Word) {
-    const varData = variables[pluginVersion.value];
-    if (varData) {
-      const currentValue = varData.value;
-      const fileReplacePosition = varData.fileReplacePosition;
-      dep.currentValue = currentValue;
-      dep.managerData = {
-        fileReplacePosition,
-        packageFile: varData.packageFile,
-      };
-    } else {
-      const currentValue = pluginVersion.value;
-      const fileReplacePosition = pluginVersion.offset;
-      dep.currentValue = currentValue;
-      dep.managerData = { fileReplacePosition, packageFile };
-      dep.skipReason = 'unknown-version';
-    }
-  } else if (pluginVersion.type === TokenType.StringInterpolation) {
-    const versionTpl = pluginVersion as StringInterpolation;
-    const children = versionTpl.children;
-    const [child] = children;
-    if (child?.type === TokenType.Variable && children.length === 1) {
-      const varData = variables[child.value];
-      if (varData) {
-        const currentValue = varData.value;
-        const fileReplacePosition = varData.fileReplacePosition;
-        dep.currentValue = currentValue;
-        dep.managerData = {
-          fileReplacePosition,
-          packageFile: varData.packageFile,
+import { isDependencyString, parseDependencyString } from './utils';
+
+const groovy = lang.createLang('groovy');
+
+const qStringValue = q.str((ctx: Ctx, node: lexer.Token) => {
+  storeVarToken(ctx, node);
+  return ctx;
+});
+
+const qVariableAssignmentIdentifier = q
+  .sym(storeVarToken)
+  .many(
+    q.alt(
+      q.op<Ctx>('.').sym(storeVarToken),
+      q.tree<Ctx>({
+        type: 'wrapped-tree',
+        maxDepth: 1,
+        startsWith: '[',
+        endsWith: ']',
+        search: q.begin<Ctx>().join(qStringValue).end(),
+      })
+    ),
+    0,
+    32
+  )
+  .handler(stripReservedPrefixFromKeyTokens);
+
+const qVariableAccessIdentifier =
+  qVariableAssignmentIdentifier.handler(coalesceVariable);
+
+const qPropertyAccessIdentifier = q
+  .alt(
+    q.sym<Ctx>('property'),
+    q
+      .sym<Ctx>(/^(rootProject|project)$/)
```

Comment:
Should work well with RE2. Also, capturing groups aren't necessary, though not sure if they're impacting perf seriously.

```suggestion
      .sym<Ctx>(regEx(/^(?:rootProject|project)$/))
```